public/pages/workflows/workflows.tsx

```diff
@@ -99,6 +106,45 @@ export function Workflows(props: WorkflowsProps) {
   const { workflows, loading } = useSelector(
     (state: AppState) => state.workflows
   );
+
+  // run health checks on FF and ML commons, any time there is a new selected datasource (or none if MDS is disabled)
+  // block all user actions if there are failures executing the basic search APIs for either plugin.
+  const [
+    flowFrameworkConnectionErrors,
+    setFlowFrameworkConnectionErrors,
+  ] = useState<boolean>(false);
+  const [mlCommonsConnectionErrors, setMLCommonsConnectionErrors] = useState<
+    boolean
+  >(false);
+  const connectionErrors =
+    flowFrameworkConnectionErrors || mlCommonsConnectionErrors;
+  useEffect(() => {
+    async function flowFrameworkHealthCheck() {
+      await dispatch(
+        searchWorkflows({
+          apiBody: FETCH_ALL_QUERY_LARGE,
+          dataSourceId,
+        })
+      ).then((resp: any) => {
+        setFlowFrameworkConnectionErrors(!isEmpty(resp.error));
+      });
+    }
+    async function mlCommonsHealthCheck() {
+      await dispatch(
+        searchModels({
+          apiBody: FETCH_ALL_QUERY_LARGE,
+          dataSourceId,
+        })
+      ).then((resp: any) => {
+        setMLCommonsConnectionErrors(!isEmpty(resp.error));
+      });
+    }
```

Comment:
nit: it seems like you're making two independent async calls here, maybe consider run them in parallel for efficiency

Reply:
Thanks for the suggestion - updated to run them inside a single `Promise.all()` list.